DOCUMENTATION.md

```diff
@@ -107,19 +122,162 @@ In case it's a **dictionary** with fields, please make sure the **encoder** func
+- `certificate`: **string**  this is the content of the certificate used to connect in a secure way to the handler. It needs to be provided as a `certificate=value` argument when calling the constructor.
+- `discovery_address`: **string**  this is the address of the discovery server to use in order to find back the address of the handler to which the application in registered. It needs to be provided as a `discovery_address=value` argument when calling the constructor.
+- `path_to_key=`: **string**  this is the absolute path to the file which contains the key from which the token you wish to use, is signed. It needs to be provided as a `path_to_key=value` argument when calling the constructor.
+The constructor returns an **ApplicationClient** object set up with the application informations, ready to get the application registered on The Things Network.
+
+### get
+Gives back the [**Application**](#application) object with the id given to the constructor.
+```python
+client.get()
+```
+
+### set_payload_format
+Sets the payload format of the application.
+```python
+client.set_payload_format(payload_format)
+```
+- payload_format: **string**  the new payload format.
+
+### set_custom_payload_functions
+Sets the payload functions of the application.
+```python
+client.set_custom_payload_functions([decoder], [encoder], [validator], [converter])
+```
+- `decoder`: **string**  decoder function that must be written in javascript, it needs to be provided as a `decoder=value` argument when calling the method.
+- `encoder`: **string**  encoder function that must be written in javascript, it needs to be provided as a `encoder=value` argument when calling the method.
+- `validator`: **string**  validator function that must be written in javascript, it needs to be provided as a `validator=value` argument when calling the method.
+- `converter`: **string**  converter function that must be written in javascript, it needs to be provided as a `converter=value` argument when calling the method.
+
+### set_register_on_join_access_key
+Sets the register on join access key of the application.
+```python
+client.set_register_on_join_access_key(register_on_join)
+```
+- `register_on_join`: **string**  the `register_on_join` access key.
+
+### unregister
+Unregisters the application of the id provided to the constructor on creation of the client.
+```python
+client.unregister()
+```
+
+### register_device
+Registers a new device to the application.
+```python
+client.register_device(dev_id, device)
+```
+- `dev_id`: **string**  the id of the device to be registered.
+- `device`: **dictionary**  the dictionary with fields to be set as a new device of the application. See the [Device](#device) section to know the structure of the dictionary that should be passed and the name of the fields.
```

Comment:
There are multiple #device anchors in the document, add a more specific Markdown anchor to specify which one you're linking to